users/api/clients.go

```diff
@@ -65,81 +64,102 @@ func clientsHandler(svc users.Service, authn mgauthn.Authentication, tokenClient
 			), "view_profile").ServeHTTP)
 
 			r.Get("/{id}", otelhttp.NewHandler(kithttp.NewServer(
-				viewClientEndpoint(svc),
-				decodeViewClient,
+				viewEndpoint(svc),
+				decodeViewUser,
 				api.EncodeResponse,
 				opts...,
-			), "view_client").ServeHTTP)
+			), "view_user").ServeHTTP)
 
 			r.Get("/", otelhttp.NewHandler(kithttp.NewServer(
-				listClientsEndpoint(svc),
-				decodeListClients,
+				listUsersEndpoint(svc),
+				decodeListUsers,
 				api.EncodeResponse,
 				opts...,
-			), "list_clients").ServeHTTP)
+			), "list_users").ServeHTTP)
 
 			r.Get("/search", otelhttp.NewHandler(kithttp.NewServer(
-				searchClientsEndpoint(svc),
-				decodeSearchClients,
+				searchUsersEndpoint(svc),
+				decodeSearchUsers,
 				api.EncodeResponse,
 				opts...,
-			), "search_clients").ServeHTTP)
+			), "search_users").ServeHTTP)
+
+			r.Get("/username", otelhttp.NewHandler(kithttp.NewServer(
+				viewByUserNameEndpoint(svc),
+				decodeViewUserByUserName,
+				api.EncodeResponse,
+				opts...,
+			), "view_user_by_username").ServeHTTP)
 
 			r.Patch("/secret", otelhttp.NewHandler(kithttp.NewServer(
-				updateClientSecretEndpoint(svc),
-				decodeUpdateClientSecret,
+				updateSecretEndpoint(svc),
+				decodeUpdateUserSecret,
 				api.EncodeResponse,
 				opts...,
-			), "update_client_secret").ServeHTTP)
+			), "update_user_secret").ServeHTTP)
 
 			r.Patch("/{id}", otelhttp.NewHandler(kithttp.NewServer(
-				updateClientEndpoint(svc),
-				decodeUpdateClient,
+				updateEndpoint(svc),
+				decodeUpdateUser,
+				api.EncodeResponse,
+				opts...,
+			), "update_user_name_and_metadata").ServeHTTP)
+
+			r.Patch("/{id}/names", otelhttp.NewHandler(kithttp.NewServer(
+				updateUserNamesEndpoint(svc),
+				decodeUpdateUserNames,
 				api.EncodeResponse,
 				opts...,
-			), "update_client").ServeHTTP)
+			), "update_user_names").ServeHTTP)
```

Comment:
Do we not have repeated functionality with this, the `update_user` can update the first and last name as well as the username and this endpoint does the same. I would think that this would be for updating specifically the `userName` . The updae_user one can update first and last name as well as the metadata, or  it can perform all the fucntionality